app/models/location.rb

```diff
@@ -15,6 +15,69 @@ def self.primary(city_name = ENV['PRIMARY_CITY_NAME'])
   def weather
     response = ::Clients::WeatherClient.forecast(self).dup
     response['moonPhase'] = moon_phase
+    
+   response['current']['weather'][0]['hourly'] = response['hourly']
+
+   weather_info = {"rain" => "response['hourly'][2]['pop']" , "snow" => "response['hourly'][2]['snow']"}
+   weather_info.select {|key,value|
+   case
+   when value.to_f >=0.25 then
+    response['precip_message'] = "possible to #{key} in next hour"
+    when value.to_f >= 0.50 then
+     response['precip_message'] = "Likely to #{key} in next hour"
+    when value.to_f >= 0.75 then
+    response['precip_message'] = "#{key} in next hour"
+    else
+      for i in 3..24 do
+      time_split = Time.at(response['hourly'][i]['dt'].to_i).strftime("%H")
+       case time_split.to_i
+        when 12..18 then 
+          weather_info = {"rain" => "response['hourly'][i]['pop']" , "snow" => "response['hourly'][i]['snow']"}
+          weather_info.select {|key,value|
+          case
+          when value.to_f >=0.25 then
+           response['precip_message'] = "possible to #{key} this afternoon"
+           when value.to_f >= 0.50 then
+            response['precip_message'] = "Likely to #{key} this afternoon"
+           when value.to_f >= 0.75 then
+           response['precip_message'] = "#{key} this afternoon"
+           else    
+           response['precip_message'] = "possible to #{key} this afternoon"
+         end
+       }
+        when 18..21 then 
+          weather_info = {"rain" => "response['hourly'][i]['pop']" , "snow" => "response['hourly'][i]['snow']"}
+          weather_info.select {|key,value|
+          case
+          when value.to_f >=0.25 then
+           response['precip_message'] = "possible to #{key} this evening"
+           when value.to_f >= 0.50 then
+            response['precip_message'] = "Likely to #{key} this evening"
+           when value.to_f >= 0.75 then
+           response['precip_message'] = "#{key} this evening"
+           else    
+           response['precip_message'] = "possible to #{key} this evening"
+         end
+       }
+        when 21..24 then 
+          weather_info = {"rain" => "response['hourly'][i]['pop']" , "snow" => "response['hourly'][i]['snow']"}
+          weather_info.select {|key,value|
+          case
+          when value.to_f >=0.25 then
+           response['precip_message'] = "possible to #{key} tonight"
+           when value.to_f >= 0.50 then
+            response['precip_message'] = "Likely to #{key} tonight"
+           when value.to_f >= 0.75 then
+           response['precip_message'] = "#{key} tonight"
+           else    
+           response['precip_message'] = "possible to #{key} tonight"
+         end
+       }  
+       end
+     end     
```

Comment:
It looks like the only difference between your various `case` statements is the time frame. It might be worth pulling out a helper method that takes `value.to_f` and a `time_frame` string to reduce some of the duplication.

Thatl'll also help to make the flow through this complicated conditional easier to understand. Right now there's at least one bug - it doesn't handle the case where it's not going to rain or snow in the next 24 hours - but it's hard to tell that as it's currently written.